### ISSUE TITLE
Websocket API

### ISSUE THREAD

**Issue by bzero** [butor/blackbird#96]
any chances to implement websocket ? there are some exchange offering it:

https://www.bitstamp.net/websocket/
https://bitfinex.readme.io/v2/docs/ws-general

the polling technique is to old slow and the timeouts are constant 

[ 01/12/2017 11:27:30 ]
   Bitfinex: 	764,29 / 764,30
Error with cURL: Timeout was reached
  URL: https://www.bitstamp.net/api/ticker/
  Retry in 2 sec...
   Bitstamp: 	764,10 / 764,96
   ----------------------------
   Bitstamp/Bitfinex:	-0.09% [target  0.80%, min -0.51%, max  1.04%]

**Comment by bzero** [comment]
`[ 01/12/2017 16:52:15 ]
   Bitfinex: 	805,47 / 805,48
Error with JSON:
'[' or '{' expected near '<'
Buffer:
<html style="height:100%"><head><META NAME="ROBOTS" CONTENT="NOINDEX, NOFOLLOW"><meta name="format-detection" content="telephone=no"><meta name="viewport" content="initial-scale=1.0"><meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1"></head><body style="margin:0px;height:100%"><iframe src="/_Incapsula_Resource?CWUDNSAI=19&xinfo=18-5658367-0 NNNN RT(1484239921177 14635) q(0 0 -1 -1) r(150 -1) b6&incident_id=0-115617126623608850&edet=20&cinfo=ffffffff068d" frameborder=0 width="100%" height="100%" marginheight="0px" marginwidth="0px">Request unsuccessful. Incapsula incident ID: 0-115617126623608850</iframe></body></html>
Retrying...
   Bitstamp: 	807,51 / 807,56
   ----------------------------
   Bitstamp/Bitfinex:	-0.26% [target  0.80%, min -0.89%, max  1.04%]

WARNING: 16 second(s) too late at 01/12/2017 16:52:18
`

**Comment by butor** [comment]
Yes, that would be a great enhancement and a good implementation example for other bitcoin projects.
